nym-vpn-core/crates/nym-vpn-lib/src/platform/account.rs

```diff
@@ -326,14 +326,54 @@ pub(crate) mod raw {
         })
     }
 
-    pub(crate) async fn forget_account_raw(path: &str) -> Result<(), VpnError> {
+    async fn unregister_device_from_api_raw(
+        path: &str,
+        network_env: Network,
+    ) -> Result<(), VpnError> {
+        let account_storage = setup_account_storage(path).await?;
+        let device_keys = account_storage
+            .load_keys()
+            .await
+            .map_err(|_| VpnError::NoDeviceIdentity)?;
+        let device = Device::from(device_keys.device_keypair().clone());
+        let mnemonic = account_storage
+            .load_mnemonic()
+            .await
+            .map_err(|_| VpnError::NoAccountStored)?;
+        let account = VpnApiAccount::from(mnemonic);
+        let user_agent = crate::util::construct_user_agent();
+
+        let vpn_api_client =
+            nym_vpn_api_client::VpnApiClient::new(network_env.vpn_api_url(), user_agent).map_err(
+                |e| VpnError::InternalError {
+                    details: e.to_string(),
+                },
+            )?;
+        vpn_api_client
+            .update_device(&account, &device, DeviceStatus::DeleteMe)
+            .await
+            .map_err(|err| VpnError::UnregisterDeviceApiClientFailure {
+                details: err.to_string(),
+            })?;
+        Ok(())
+    }
+
+    pub(crate) async fn forget_account_raw(path: &str, network: &str) -> Result<(), VpnError> {
         tracing::info!("REMOVING ALL ACCOUNT AND DEVICE DATA IN: {path}");
 
+        let network_env = nym_vpn_network_config::Network::fetch(network).map_err(|err| {
```

Comment:
Get the environment from `current_environment` instead, so you don't have to manually fetch it. And also risk having a network mismatch